### DISCUSSION TITLE
anybody having issues with using the grid on a non-primary monitor

### DISCUSSION THREAD

**Discussion by jf** [agzam/spacehammer#200] [General]
As an example, on my main monitor, doing "w s" places a window into the middle column of the screen, but on my secondary, it always places it where "q a" is instead.

**Comment by Grazfather** [comment]
I don't, but I noticed that I need my `hs.grid.HINTS to have one more row than my grid setting.

```fennel
;; Make the grid hints look like my keyboard layout
(set hs.grid.HINTS
     [[ "f1" "f2" "f3" "f4" "f5" "f6" "f7" "f8" "f9" "f10" ] ; This line doesn't show
      [ "1" "2" "3" "4" "5" "6" "7" "8" "9" "0" ]
      [ "Q" "W" "F" "P" "V" "J" "L" "U" "Y" ";" ]
      [ "A" "R" "S" "T" "G" "M" "N" "E" "I" "O" ]
      [ "Z" "X" "C" "D" "B" "K" "H" "," "." "/" ]])

(local config
       {:title "Main Menu"
        ;...
        :grid {:size "9x4"}
        :hyper {:key :F18}})
```

**Reply by jf** [reply]
`hs.inspect(hs.grid.HINTS)` shows this, so I think i'm good?

```
> hs.inspect(hs.grid.HINTS)
{ { "f1", "f2", "f3", "f4", "f5", "f6", "f7", "f8", "f9", "f10" }, { "1", "2", "3", "4", "5", "6", "7", "8", "9", "0" }, { "Q", "W", "E", "R", "T", "Y", "U", "I", "O", "P" }, { "A", "S", "D", "F", "G", "H", "J", "K", "L", ";" }, { "Z", "X", "C", "V", "B", "N", "M", ",", ".", "/" } }
```

**Reply by Grazfather** [reply]
what is your grid size in your spacehammer config?
I think it should be 10x4 in your case.

**Reply by jf** [reply]
I started out with 3x2 as per spacehammer's default in `window.fnl`.

Do the hints have to match up with the grid size? If that's the case that just seems like an awful way to have to configure things. I was thinking that `hs.grid` would simply take as many hints from the table as necessary. So in the case of 3x2, it'll just do Q, W, E, then A, S, D (which it does). And then I when I update the grid to 3x3, I get an additional row showing Z, X, C. Is this not how the hints are supposed to work?

**Reply by Grazfather** [reply]
I'm not sure, I just remember having to get them inline for normal behaviour

**Reply by jf** [reply]
ok. I just tried it to be sure, and it does not fix things for me. Maybe something else to dig into / to add to my list of things to dig into. Thank you regardless!
